app/services/user/student_service.py

```diff
@@ -47,16 +47,60 @@ async def create_student(
         course_service = CourseService(self.session)
 
         master_repo_name = await course_service.get_master_repository_name()
+        student_repo_name = await course_service.get_student_repository_name(onyen)
         instructor_organization = await course_service.get_instructor_gitea_organization_name()
+        
         await gitea_service.create_user(onyen, email, password)
+        await gitea_service.add_collaborator_to_repo(
+            name=master_repo_name,
+            owner=instructor_organization,
+            collaborator_name=onyen,
+            permission=CollaboratorPermission.READ
+        )
         await gitea_service.fork_repository(
             name=master_repo_name,
             owner=instructor_organization,
             new_owner=onyen
         )
+        # The remote is subject to change when renamed, so we don't use the remote returned by fork_repository.
+        student.fork_remote_url = await gitea_service.modify_repository(
+            name=master_repo_name,
+            owner=onyen,
+            new_name=student_repo_name
+        )
+        self.session.commit()
+
+    async def delete_user(
+        self,
+        onyen: str
+    ) -> None:
+        from app.services import SubmissionService, AssignmentService
+        from app.models import ExtraTimeModel
```

Comment:
I copied over the delete user in student service & user service into my PR. Just to keep in mind when we merge them both